File: 2559-count-vowel-strings-in-ranges/2559-count-vowel-strings-in-ranges.py
class Solution:
    def vowelStrings(self, words: List[str], queries: List[List[int]]) -> List[int]:
        prefix_vowels, ans = [0] * (len(words) + 1), [0] * len(queries)
        vowels = {'a', 'e', 'i', 'o', 'u'}

        for i in range(len(words)):
            if not (words[i][0] in vowels and words[i][-1] in vowels):
                prefix_vowels[i + 1] = prefix_vowels[i]
            else:
                prefix_vowels[i + 1] = prefix_vowels[i] + 1

        for i, (left, right) in enumerate(queries):
            ans[i] = prefix_vowels[right + 1] - prefix_vowels[left]

        return ans

File: 2559-count-vowel-strings-in-ranges/2559-count-vowel-strings-in-ranges.cpp
class Solution {
public:
    vector<int> vowelStrings(vector<string>& words, vector<vector<int>>& queries) {
        vector<int> prefixVowels(words.size() + 1, 0), ans(queries.size(), 0);
        unordered_set<char> vowels{'a', 'e', 'i', 'o', 'u'};

        for (int i = 0; i < words.size(); ++i){
            if (!(vowels.count(words[i][0]) && vowels.count(words[i].back())))
                prefixVowels[i + 1] = prefixVowels[i];
            else
                prefixVowels[i + 1] = prefixVowels[i] + 1;
        }

        for (int i = 0; i < queries.size(); ++i){
            int left = queries[i][0], right = queries[i][1];
            ans[i] = prefixVowels[right + 1] - prefixVowels[left];
        }

        return ans;
    }
};